### ISSUE TITLE
MW 1.32 `ContentLanguage` service causes test to fail

### ISSUE THREAD

**Issue by mwjames** [SemanticMediaWiki/SemanticMediaWiki#3300]
### Setup and configuration

- SMW version:  all
- MW version: 1.32+
- PHP version: all
- DB system (MySQL, Blazegraph, etc.) and version: all

### Issue

Beginning with [0] tests do fail with "InvalidArgumentException: No matchable hierarchy type..." due to a missing identification of the correct NS_CATEGORY NS assignment which is assumed to hold a modified `wgContLang` (tests that fail against an `es` content language) but since `wgContLang` can no longer be modified directly (replaced by the `ContentLanguag` service), the `Title` creation with a different content language other than that of the invoked language (meaning that from the request start) causes the `Title` parsing to fail and eventually doesn't recognize the `es` NS_CATEGORY as namespace.

As symptom, `HierarchyLookup` fails because it expects a NS_CATEGORY entity.

### Stack trace

Before [0]:

- Before: https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/409714050
- After: https://travis-ci.org/SemanticMediaWiki/SemanticMediaWiki/jobs/411177230

```
2) SMW\Tests\Integration\JSONScript\JsonTestCaseScriptRunnerTest::testCaseFile with data set "p-0438.json" ('/home/travis/build/SemanticMe...8.json')
InvalidArgumentException: No matchable hierarchy type.
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/HierarchyLookup.php:217
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/HierarchyLookup.php:162
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/SQLStore/QueryDependency/QueryResultDependencyListResolver.php:164
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/SQLStore/QueryDependency/QueryResultDependencyListResolver.php:115
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/SQLStore/QueryDependency/QueryDependencyLinksStore.php:397
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/SQLStore/QueryDependency/QueryDependencyLinksStore.php:360
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/MediaWiki/Deferred/CallableUpdate.php:225
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/src/MediaWiki/Deferred/TransactionalCallableUpdate.php:140
/home/travis/build/SemanticMediaWiki/mw/includes/deferred/DeferredUpdates.php:268
/home/travis/build/SemanticMediaWiki/mw/includes/deferred/DeferredUpdates.php:226
/home/travis/build/SemanticMediaWiki/mw/includes/deferred/DeferredUpdates.php:130
/home/travis/build/SemanticMediaWiki/mw/includes/deferred/DeferredUpdates.php:307
/home/travis/build/SemanticMediaWiki/mw/includes/deferred/DeferredUpdates.php:100
/home/travis/build/SemanticMediaWiki/mw/includes/Storage/PageUpdater.php:1163
/home/travis/build/SemanticMediaWiki/mw/includes/Storage/PageUpdater.php:730
/home/travis/build/SemanticMediaWiki/mw/includes/page/WikiPage.php:1855
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/Utils/PageCreator.php:93
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/Utils/PageCreator.php:71
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/JsonTestCaseContentHandler.php:149
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/JsonTestCaseContentHandler.php:112
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/JsonTestCaseScriptRunner.php:297
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/Integration/JSONScript/JsonTestCaseScriptRunnerTest.php:319
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/Integration/JSONScript/JsonTestCaseScriptRunnerTest.php:201
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/JsonTestCaseScriptRunner.php:204
/home/travis/build/SemanticMediaWiki/mw/extensions/SemanticMediaWiki/tests/phpunit/DatabaseTestCase.php:133
/home/travis/build/SemanticMediaWiki/mw/maintenance/doMaintenance.php:94
```

### Steps to reproduce

[0] https://github.com/wikimedia/mediawiki/commit/49ce67be93dfbb40d036703dad2278ea9843f1ad

**Comment by mwjames** [comment]
We may be able to circumvent the issue but finding the culprit, reading the git log, and interpreting the appropriate line of action has taken again 2h of time that could have spent more productively towards the project.
